js/ui/appIconBar.js

```diff
@@ -657,6 +662,8 @@ const ScrolledIconList = GObject.registerClass({
                 appButton.add_style_pseudo_class('highlighted');
             else
                 appButton.remove_style_pseudo_class('highlighted');
+
+            appButton.queue_redraw();
```

Comment:
As discussed on Slack with @GeorgesStavracas I was actually trying to fix this issue on `st_widget_recompute_style()` based on [this comment](https://github.com/endlessm/gnome-shell/pull/497#discussion_r303653166), but @GeorgesStavracas suggested we go with this as we won't have a solution for `st_widget_recompute_style()` for now.

This is fine to me and the changes work fine on local testing, so merging.